tests/integration/test_dedup_reporter.py

```
@@ -0,0 +1,112 @@
+    assert out_file.exists()
+    parsed = json.loads(out_file.read_text(encoding="utf-8"))
+    assert parsed["num_groups"] == 2
+
+
+def test_storage_reporter_export_to_csv_oserror(
+    tmp_path: Path, sample_duplicate_groups: list[dict[str, Any]]
+) -> None:
+    reporter = StorageReporter()
+    out_file = tmp_path / "report.csv"
+    with patch("builtins.open", side_effect=OSError("disk full")):
+        with pytest.raises(OSError):
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)